graph/appender/unused_service_test.go

```diff
@@ -113,9 +113,9 @@ func TestVersionWithNoTrafficScenario(t *testing.T) {
 	assert.Equal("true", recommendationV2.Metadata["isUnused"])
 }
 
-func mockDeploments() *v1beta1.DeploymentList {
-	deployments := v1beta1.DeploymentList{
-		Items: []v1beta1.Deployment{
+func mockPods() *v1.PodList {
```

Comment:
Suggestion, as we introduced pods that might have a 1:N relationship, I would duplicate same pods with same app x label to check that the new check works fine.

Reply:
+1, it will be expected to have multiple pods for the same service+version.

Reply:
Done!